xrally_kubernetes/service.py

```diff
@@ -15,9 +15,90 @@
 from kubernetes.client import api_client
 from kubernetes.client.apis import core_v1_api
 from kubernetes.client.apis import version_api
+from kubernetes.client import rest
+from rally.common import cfg
+from rally.common import logging
+from rally.common import utils as commonutils
+from rally import exceptions
 from rally.task import atomic
 from rally.task import service
 
+CONF = cfg.CONF
+LOG = logging.getLogger(__name__)
+
+
+def wait_for_status(name, status, read_method, **kwargs):
+    """Util method for polling status until it won't be equals to `status`.
+
+    :param name: resource name
+    :param status: status waiting for
+    :param read_method: method to poll
+    :param kwargs: additional kwargs for read_method
+    """
+    sleep_time = CONF.kubernetes.status_poll_interval
+    retries_total = CONF.kubernetes.status_retries_timeout
+
+    commonutils.interruptable_sleep(CONF.kubernetes.start_prepoll_delay)
+
+    i = 0
+    resp_id = None
+    current_status = None
+    while i < retries_total:
```

Comment:
It looks incorrect. The variable `i` contains the number of attempts and you are comparing it with a number in seconds. You need to use start and current timestamps to calculate the elapsed time. 

Also, L62 is wrong, since read operation can take seconds in case of bad connectivity.

PS: the same comments to wait_for_not_found method.

Reply:
`retries_total` is total number of attempts, not total number of seconds.

Reply:
"Kubernetes status retries total timeout" <- help message for an option. "timeout" usually describes time.
Also, time is more general for other plugins what we have.